packages/sha1-browser/src/ie11Sha1.ts

```diff
@@ -1,34 +1,25 @@
 import { isEmptyData } from "./isEmptyData";
 import { SHA_1_HMAC_ALGO } from "./constants";
-import { Hash, SourceData } from "@aws-sdk/types";
+import { Checksum, SourceData } from "@aws-sdk/types";
 import { fromUtf8 } from "@aws-sdk/util-utf8-browser";
 import { CryptoOperation, Key, MsWindow } from "@aws-crypto/ie11-detection";
 import { locateWindow } from "@aws-sdk/util-locate-window";
 
-export class Sha1 implements Hash {
-  private operation: Promise<CryptoOperation>;
+export class Sha1 implements Checksum {
+  private secret?: SourceData;
+  private operation?: Promise<CryptoOperation>;
```

Comment:
Suggestion: Use the [Definite Assignment Operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html#definite-assignment-assertions) to tell Type Script that `operation` will be assigned.
```suggestion
  private operation!: Promise<CryptoOperation>;
```

This will let you remove the `@ts-ignore`s below.